Add some configuration options to allow either the Analog reading of the FSR, or digital reading

tmolteno opened the pull request:
Hi There

I have made a couple of changes.

1) Configuration.h has had a few defines added to allow either FSR-analog or FSR-digital.
2) qr_solve has had an include changed to allow compilation
3) Marlin_main has had the code cleaned up around reading the state of the bed touching in the auto-callibration.

Kind Regards

Tim Molteno

tmolteno committed:
Changed include to allow compilation

tmolteno committed:
Added series of defines to allow either the FSR to be read by an analog input, or as a digital input (ZMIN)

cdaringe commented:
It looks like some of your personal config made it into this pull req.  It may be worth saving the diff, re-checking out, and commiting only your features/updates

tmolteno commented:
Good point. Apologies for that. The problem is that the commits will then be untested  (as I can't run them on my machine). 

How about I refactor the configuration.h into several configuration files. One for machine geometry, another for bed leveling e.t.c

configuration.h, (most of the usual stuff)
    config_geometry.h (dimensions entered during calibration)
    config_autocal.h (fsr configurations e.t.c)

This will make it easier to work from a git repository.

P.S. Should the xyz pulley size default (XYZ_PULLEY_TEETH) really be 17 teeth? Mine have 16.

Cheers
Tim

tmolteno committed:
Added a new file that contains all the usual calibration inputs for a typical delta printer. This makes hunting for the configuration options, and setting them a little easier for the novice user. It also makes it easier to develop Marlin firmware without spamming the main configuration.h with local changes.

tmolteno committed:
Include the local calibration file, and modify the order of includes to allow the most common calibrations to be stored in Calibration.h.

tmolteno committed:
Removed the VPATH for the Time library. It isnt used

tmolteno commented:
Hi There

I have make the changes I suggested in the last comment. 

1) Tried to put all the sensible local overrides into a file called Calibration.h. 
2) Added a comment to note that the existing defaults are suitable for a Kossel Air (rather than a Mini)
3) Added a new SCALING_FACTOR option that is modified when printing test cubes to calibrate dimensions.
4) Added an EXTRUDER_STEPS option that is modified when calibrating the extruder.

Now the pull request includes, both those changes, and the cleanup of the FSR code to allow either ZMIN or TEMP inputs to be used for auto bed leveling, as described in by earlier comments.

Happy to receive suggestions for improvements.

Kind Regards

Tim

tmolteno committed:
Set the build radius for a Kossel Mini. Add instructions for options not in this file

tmolteno committed:
Updated to allow simultaneous heated bed and use the extra thermometer input to read the force sensitive resistors for the level sensing and auto callibration

tmolteno committed:
Fix up new z distance
